modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1830,11 +1830,22 @@ private CompletableFuture<FuturesCleanupResult> awaitCleanupReadyFutures(UUID tx
         List<CompletableFuture<?>> txUpdateFutures = new ArrayList<>();
         List<CompletableFuture<?>> txReadFutures = new ArrayList<>();
 
+        AtomicBoolean forceCleanup = new AtomicBoolean(true);
+
         txCleanupReadyFutures.compute(txId, (id, txOps) -> {
             if (txOps == null) {
                 return null;
             }
 
+            // Update futures are empty in two cases:
+            // - there were no actions in the transaction
```

Comment:
Formally it's not true, update futures will be empty if  "there were no **write** actions in the transaction". However, the logic itself is correct because 'txOps.futures' is used instead of updateFutures. So, please, adjust the javadoc.